docstore/driver/document.go

```diff
@@ -182,3 +182,14 @@ func (d Document) Decode(dec Decoder) error {
 	}
 	return decodeStruct(d.s, dec)
 }
+
+// RevisionOn returns whether or not we should write a revision value when we
+// write a document. It could be either a map or a struct with a revision key or
+// field named after revField.
+func (d Document) RevisionOn(revField string) bool {
+	if d.m != nil {
+		_, ok := d.m[revField]
+		return ok
+	}
+	return d.fields.MatchFold(revField) != nil
```

Comment:
This will break if you weren't in lowercase mode had two fields, `Revision` and `revision`. (You probably shouldn't do that, but we do say we'll match everything exactly except for mongo lowercase mode, so we should do that.) 

I think this should do an exact match, and maybe in the mongo driver, when you call it, you lowercase the argument to it first. I think `mongodocstore.collection.revisionField` is the right version of the field name to use.

Reply:
That won't work, `collection.revisionField` is the right version in the DB, not necessarily the right one for the user-provided document, which is the `d` here.

The easiest way is to have another `HasFieldFold` for such use case, which is what I choose for now. The advantage is that it is easy to implement and less confusion for the user to use.

Another way I've tried is to store the field user passed in and the lower them when we need to (e.g. encoding, writing) but they all end up very confusing both for implementing and using the API with lowercaseFields on.